core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/browsebars/BrowseBarComplete.java

```diff
@@ -85,141 +86,167 @@ private String printBreadCrumb() {
+        href = "#";
+      }
+
+      if (breadcrumb.length() > 0) {
+        breadcrumb.append(CONNECTOR);
+      }
+      breadcrumb.append("<a href=\"").append(href).append("\"");
+      breadcrumb.append(" class=\"space\"");
+      breadcrumb.append(" id=\"space").append(spaceId).append("\"");
+      breadcrumb.append(">");
+      breadcrumb.append(Encode.forHtml(spaceInst.getName(getUserLanguage())));
+      breadcrumb.append("</a>");
+    }
+    return breadcrumb.toString();
+  }
+
+  private String getComponent(SilverpeasComponentInstance componentInst) {
+    StringBuilder breadcrumb = new StringBuilder();
+    if (componentInst != null) {
+      // Display component's label
+      if (!componentInst.isPersonal()) {
+        breadcrumb.append(CONNECTOR);
+      }
+      breadcrumb.append("<a href=\"");
+      if (!isClickable()) {
+        breadcrumb.append("#");
+      } else if (StringUtil.isDefined(getComponentJavascriptCallback())) {
+        breadcrumb.append("javascript:").append(getComponentJavascriptCallback()).append("('")
+            .append(getComponentId()).append("')");
+      } else {
+        breadcrumb.append(URLUtil.getApplicationURL()).append(URLUtil.getURL(getSpaceId(),
+            getComponentId()));
+        if (ignoreComponentLink()) {
+          breadcrumb.append("Main");
+        } else {
+          breadcrumb.append(getComponentLink());
+        }
+      }
+      breadcrumb.append("\"");
+      breadcrumb.append(" class=\"component\"");
+      breadcrumb.append(" id=\"bc_").append(componentInst.getId()).append("\"");
+      breadcrumb.append(">");
+      breadcrumb.append(Encode.forHtml(componentInst.getLabel(getUserLanguage())));
+      breadcrumb.append("</a>");
+    }
+    return breadcrumb.toString();
+  }
+
+  private void appendPath(StringBuilder breadcrumb) {
     List<BrowseBarElement> elements = getElements();
     if (!elements.isEmpty()) {
       for (BrowseBarElement element : elements) {
-        if (!emptyBreadCrumb) {
-          result.append(CONNECTOR);
+        if (breadcrumb.length() > 0) {
+          breadcrumb.append(CONNECTOR);
+        }
+        if (StringUtil.isDefined(element.getLink())) {
+          breadcrumb.append("<a href=\"").append(element.getLink()).append("\"");
+        } else {
+          breadcrumb.append("<span");
         }
-        result.append("<a href=\"").append(element.getLink()).append("\"");
-        result.append(" class=\"element\"");
+        breadcrumb.append(" class=\"element\"");
         if (StringUtil.isDefined(element.getId())) {
-          result.append(" id=\"").append(element.getId()).append("\"");
+          breadcrumb.append(" id=\"").append(element.getId()).append("\"");
+        }
+        breadcrumb.append(">");
+        breadcrumb.append(WebEncodeHelper.javaStringToHtmlString(element.getLabel()));
+        if (StringUtil.isDefined(element.getLink())) {
+          breadcrumb.append("</a>");
+        } else {
+          breadcrumb.append("</span>");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "</span>" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)